src/node.cc

```diff
@@ -447,17 +447,37 @@ void LoadEnvironment(Environment* env) {
   USE(StartMainThreadExecution(env));
 }
 
+#ifdef __POSIX__
+typedef void (*sigaction_cb)(int signo, siginfo_t* info, void* ucontext);
+#endif
 #if NODE_USE_V8_WASM_TRAP_HANDLER
-static std::atomic<void (*)(int signo, siginfo_t* info, void* ucontext)>
-    previous_sigsegv_action;
+static std::atomic<sigaction_cb> previous_sigsegv_action;
+
+constexpr uintptr_t kResetHandlerTag = 1;
 
 void TrapWebAssemblyOrContinue(int signo, siginfo_t* info, void* ucontext) {
   if (!v8::TryHandleWebAssemblyTrapPosix(signo, info, ucontext)) {
-    auto prev = previous_sigsegv_action.load();
+    sigaction_cb prev = previous_sigsegv_action.load();
     if (prev != nullptr) {
+      uintptr_t prev_address = reinterpret_cast<uintptr_t>(prev);
+      if (prev_address & kResetHandlerTag) {
+        // Not exactly an atomic switch, but this seems like it will have to do.
+        previous_sigsegv_action.store(nullptr);
+
+        prev_address &= ~kResetHandlerTag;
+        prev = reinterpret_cast<sigaction_cb>(prev_address);
+      }
+
       prev(signo, info, ucontext);
     } else {
       uv_tty_reset_mode();
+
+      // Reset to the default signal handler, i.e. cause a hard crash.
+      struct sigaction sa;
+      memset(&sa, 0, sizeof(sa));
+      sa.sa_handler = SIG_DFL;
+      CHECK_EQ(sigaction(signo, &sa, nullptr), 0);
```

Comment:
Do this before the call to `uv_tty_reset_mode()`? Not that that code isn't infallible and objectively perfect but you know, just in case...

Reply:
Done!